File: MiniGame-Java/src/main/java/rs/lab/mges/engine/Control.java
package rs.lab.mges.engine;

import io.github.libsdl4j.api.event.SDL_Event;
import io.github.libsdl4j.api.event.SDL_EventType;
import io.github.libsdl4j.api.keycode.SDL_Keycode;

public final class Control {

    public static boolean Left;
    public static boolean Right;
    public static boolean Up;
    public static boolean Down;
    public static boolean Fire;
    public static boolean Exit;

    public static void Reset() {
        Left = false;
        Right = false;
        Up = false;
        Down = false;
        Fire = false;
        Exit = false;
    }

    public static void HandleEvent(SDL_Event evt) {
        switch (evt.type) {
            case SDL_EventType.SDL_KEYDOWN -> {
                if (evt.key.keysym.sym == SDL_Keycode.SDLK_LEFT) {
                    Left = true;
                } else if (evt.key.keysym.sym == SDL_Keycode.SDLK_RIGHT) {
                    Right = true;
                } else if (evt.key.keysym.sym == SDL_Keycode.SDLK_UP) {
                    Up = true;
                } else if (evt.key.keysym.sym == SDL_Keycode.SDLK_DOWN) {
                    Down = true;
                } else if (evt.key.keysym.sym == SDL_Keycode.SDLK_Z) {
                    Fire = true;
                } else if (evt.key.keysym.sym == SDL_Keycode.SDLK_ESCAPE) {
                    Exit = true;
                }
            }
            case SDL_EventType.SDL_KEYUP -> {
                if (evt.key.keysym.sym == SDL_Keycode.SDLK_LEFT) {
                    Left = false;
                } else if (evt.key.keysym.sym == SDL_Keycode.SDLK_RIGHT) {
                    Right = false;
                } else if (evt.key.keysym.sym == SDL_Keycode.SDLK_UP) {
                    Up = false;
                } else if (evt.key.keysym.sym == SDL_Keycode.SDLK_DOWN) {
                    Down = false;
                } else if (evt.key.keysym.sym == SDL_Keycode.SDLK_Z) {
                    Fire = false;
                } else if (evt.key.keysym.sym == SDL_Keycode.SDLK_ESCAPE) {
                    Exit = false;
                }
            }
        }
    }
}

File: MiniGame-Python/src/engine/Control.py
from sdl2 import * # type: ignore

__all__ = ['Control']


class Control:
    Left = False
    Right = False
    Up = False
    Down = False
    Fire = False
    Exit = False

    @staticmethod
    def reset():
        Control.Left = False
        Control.Right = False
        Control.Up = False
        Control.Down = False
        Control.Fire = False
        Control.Exit = False


    @staticmethod
    def handleEvent(event):
        if event.type == SDL_KEYDOWN:
            if event.key.keysym.sym == SDLK_LEFT:
                Control.Left = True
            elif event.key.keysym.sym == SDLK_RIGHT:
                Control.Right = True
            elif event.key.keysym.sym == SDLK_UP:
                Control.Up = True
            elif event.key.keysym.sym == SDLK_DOWN:
                Control.Down = True
            elif event.key.keysym.sym == SDLK_z:
                Control.Fire = True
            elif event.key.keysym.sym == SDLK_ESCAPE:
                Control.Exit = True
        elif event.type == SDL_KEYUP:
            if event.key.keysym.sym == SDLK_LEFT:
                Control.Left = False
            elif event.key.keysym.sym == SDLK_RIGHT:
                Control.Right = False
            elif event.key.keysym.sym == SDLK_UP:
                Control.Up = False
            elif event.key.keysym.sym == SDLK_DOWN:
                Control.Down = False
            elif event.key.keysym.sym == SDLK_z:
                Control.Fire = False
            elif event.key.keysym.sym == SDLK_ESCAPE:
                Control.Exit = False
